app/test/main_test.dart

```diff
@@ -1,18 +1,24 @@
-import 'package:app/core/services/service_locator.dart';
+import 'package:app/core/services/navigation/nav_service.dart';
+import 'package:app/core/view_models/welcome_view_model.dart';
 import 'package:app/humane_transport_app.dart';
+import 'package:app/ui/views/welcome/welcome.dart';
 import 'package:flutter_test/flutter_test.dart';
+import 'package:get_it/get_it.dart';
+
+import 'test_lib/test_service_locator.dart';
+
+final testLocator = GetIt.instance;
 
 void main() {
-  group('Group test', () {
-    setUpAll(() {
-      // TODO: Make a test locator
-      setUpLocator();
-    });
+  group('Application main', () {
+    tearDown(() => resetForTest(testLocator));
 
-    testWidgets('HumaneTransportApp says Welcome', (WidgetTester tester) async {
+    testWidgets('first route is welcome screen', (WidgetTester tester) async {
+      addLazySingletonForTest(testLocator, () => WelcomeViewModel());
```

Comment:
Each test should set up only the dependencies it needs. 

Normally, we would mock these, but mocking widgets isn't necessary unless they have logic (which ours don't because we separate that into the ViewModels) and `WelcomeViewModel` has no logic right now (I'm cheap). Similarly with `NavigationService`, it's so simple that the mock would be the same.

However, we should probably inject `AppRouteGenerator` in the future and mock it.
 
 _This is more of an integration test, using both `HumaneTransportApp()` and `Welcome`, if we mocked `ARG` then all we would be testing is that `HTA` calls on it, which is the promise of `MaterialApp(onGenerateRoute: ...` and makes for a pretty dumb test._